frontend/appflowy_tauri/src/appflowy_app/hooks/notification.hooks.ts

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable no-redeclare */
+import { useEffect } from 'react';
+import { listen } from '@tauri-apps/api/event';
+import { Ok, Err, Result } from 'ts-results';
+import { SubscribeObject } from '@/services/backend/models/flowy-notification';
+import { FlowyError } from '@/services/backend/models/flowy-error';
+import {
+  DatabaseFieldChangesetPB,
+  DatabaseNotification,
+  FieldPB,
+  GroupChangesPB,
+  GroupRowsNotificationPB,
+  ReorderAllRowsPB,
+  ReorderSingleRowPB,
+  RowsChangePB,
+  RowsVisibilityChangePB,
+} from '@/services/backend';
+
+const NotificationPBMap = {
+  [DatabaseNotification.DidUpdateViewRowsVisibility]: RowsVisibilityChangePB,
+  [DatabaseNotification.DidUpdateViewRows]: RowsChangePB,
+  [DatabaseNotification.DidReorderRows]: ReorderAllRowsPB,
+  [DatabaseNotification.DidReorderSingleRow]:ReorderSingleRowPB,
+  [DatabaseNotification.DidUpdateFields]:DatabaseFieldChangesetPB,
+  [DatabaseNotification.DidGroupByField]:GroupChangesPB,
+  [DatabaseNotification.DidUpdateNumOfGroups]:GroupChangesPB,
+  [DatabaseNotification.DidUpdateGroupRow]: GroupRowsNotificationPB,
+  [DatabaseNotification.DidUpdateField]: FieldPB,
+  [DatabaseNotification.DidUpdateCell]: null,
+};
+
+type NotificationMap = typeof NotificationPBMap;
+
+type NotificationEnum = keyof NotificationMap;
+
+type NullableInstanceType<K extends ((abstract new (...args: any) => any) | null)> = K extends (abstract new (...args: any) => any) ? InstanceType<K> : void;
+
+type NotificationHandler<K extends NotificationEnum> = (result: Result<NullableInstanceType<NotificationMap[K]>, FlowyError>) => void;
+
```

Comment:
It should have some documentation here.

Reply:
Documentation has been added like that.

by the way, for cleaner, I change a bit code and replace the `NotificationEnum` with `DatabaseNotification`.